pages/common/cmake.md

```diff
@@ -1,13 +1,21 @@
 # cmake
 
-> Cross-platform build system generator.
-> It generates Makefiles, Visual Studio projects or others, depending on the target system.
+> Cross-platform build, testing and packaging automation system.
+> CMake uses its own shell-like syntax and generates recipes for many build systems.
 > More information: <https://cmake.org/cmake/help/latest/manual/cmake.1.html>.
 
-- Generate a Makefile and use it to compile a project in the same directory as the source:
+- Generate a build recipe in the current directory with `CMakeLists.txt` from a project directory:
 
-`cmake && make`
+`cmake {{path/to/project_directory}}`
 
-- Generate a Makefile and use it to compile a project in a separate "build" directory (out-of-source build):
+- Build artifacts with the generated recipe in `build_directory`:
 
-`cmake -H. -B {{build}} && make -C {{build}}`
+`cmake --build {{path/to/build_directory}}`
+
+- Install the build artifacts into /usr/local/ striping debugging symbols:
```

Comment:
Looks much better @bergentroll 👍🏻

Again, I believe @navarroaxel would prefer removing the trailing slash:
```suggestion
- Install the build artifacts into `/usr/local` striping debugging symbols:
```

Reply:
Also, since CMake is emphatically "cross-platform", could you please add the path that artifacts on Windows are installed to (if any, there may not be), in a form like:
```
- Install the build artifacts into `/usr/local` (`...` on Windows) striping debugging symbols:
```

Reply:
Oops! I noticed a typo.  I edited the above suggestion to account for that.

Reply:
> Install the build artifacts into `/usr/local` (`...` on Windows) striping debugging symbols:

I confused. Actually the default value for Microsoft Windows is `c:/Program Files/${PROJECT_NAME}`, but the example points exactly on ’/usr/local` to introduce the `--prefix` key.

Reply:
Then how about just changing it like this (if you apply this suggestion don't apply the one above):
```suggestion
- Install the build artifacts into `/usr/local` (`C:\Program Files\${PROJECT_NAME}` on windows), stripping debugging symbols:
```

Reply:
So what about this?

```Markdown
Install the build artifacts into a `prefix` directory (default is `/usr/local/` for POSIX systems and `C:\Program Files\${PROJECT_NAME}\` for Microsoft Windows), stripping debugging symbols:

`cmake --install {{path/to/build_directory}} --strip --prefix {{path/to/prefix}}`
```

Reply:
Hi, looks good, except we can make it a little shorter:
```
Install the build artifacts into a `prefix` directory (default is `/usr/local/` for POSIX systems and `C:\Program Files\${PROJECT_NAME}\` for Windows), stripping debug symbols:
```
1. "Microsoft Windows" is unnecessarily formal
2. "debug" -> "debugging" because they convey the same meaning, and I think the two "ing"s (in `stripping debugging`) sound slightly unnatural

Reply:
I dislike the point 1, because "Windows" is to general word. But I obey.

Reply:
Sorry to interject late in the discussion, but here I would apply tldr-pages' principle of non-exhaustiveness: we should prioritize clarity and conciseness, rather than attempting to cover all bases (if that makes things too verbose, which I think it does in this case).

In this case, I think it would be acceptable to simply omitting what the default prefix is, especially since this is an example of passing a custom prefix, so the default one would not be used anyway.

Reply:
That is true. But also [we should follow good practices](https://github.com/tldr-pages/tldr/pull/4378#discussion_r498453046) . And there is good practice to install binaries into `/usr/local/` (not to `/usr/` or `/opt/` or whatever). I have no idea how to show the `--prefix` key, say word about good POSIX prefix and word about Microsoft Windows prefix, and also keep generality of example. I'd prefer to skip the note of defaults and set `--prefix=/usr/local/`.

Reply:
Hrm, this seems to have gotten out of hand. Isn't it as simple as fixing the typo here?
```suggestion
- Install the build artifacts into /usr/local/ stripping debugging symbols:
```

Not sure I understand the confusion here.

Reply:
Not quite that simple @sbrl, we're also not sure whether or not to end the directory with a slash etc...

Here's what I think, based on all of this feedback:
```suggestion
- Install the build artifacts into the default directories, stripping debug symbols:
```
@sbrl @waldyrious @bergentroll looks OK to you?

Reply:
@bl-ue, so example will be without prefix nevertheless?
```Markdown
`cmake --install {{path/to/build_directory}} --strip`
```

Also "default directory" is not suitable, because the prefix is only prefix, and further paths depend on project setup at the end. If it is difficult concept, we may just purge it. It is good to say about prefixes because they especially useful on maintenment tasks like building packages in chroot environments.

Reply:
@bergentroll there are actually net even 8 examples, so I say:
1. One example showing how to install with the _default_ prefix,
2. Another example showing how to install with a _custom_ prefix.

Probably both should have `--strip` since installing usually happens for non-debug (aka release) builds

The descriptions could be something like:
```
- Install the build artifacts into the default directories, stripping debug symbols:

`...`

- Install the build artifacts into a custom directory:

`...`
```
I wouldn't think it's neccessary to specify "stripping debug symbols" twice as a user would mentally match `--strip` with the "stripping debug symbols" in the first example and see `--strip` in the next example and mentally apply it there.

What do you think?